feat: adding casestudies, blogs, tsc, roadmap, events, ambassadors page, tools page, dashboard and community page

Shriya-Chauhan opened the pull request:
This PR introduces several new pages and updates across the site to enhance navigation and improve overall user experience.

🔧 Changes Included

**Added new pages:**

- Case Studies
- Blogs
- Technical Steering Committee (TSC)
- Roadmap
- Events
- Ambassadors
- tools page
- community
- dashboard

**Fixes and updates:**

- Fixed mobile and desktop layout issues
- Updated heading sizes for better hierarchy and consistency

- Redesigned Roadmaps page

- Updated Ambassadors page content and layout

- Improved Blog desktop design

- Added new graphics and assets where needed

related issues : #3669 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added comprehensive dark mode support across the entire platform
  * Launched new community events and updates hub with integrated calendar
  * Redesigned case studies page with interactive carousel showcase
  * Added blog post filtering and pagination with category tabs
  * Introduced community tooling showcase section

* **Bug Fixes & Improvements**
  * Enhanced community pages with improved search and filtering capabilities

* **Documentation**
  * Updated Code of Conduct to version 3.0 with refined enforcement guidelines

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Shriya-Chauhan committed:
Update if-nodejs-pr-testing.yml

Shriya-Chauhan committed:
Revert "Update if-nodejs-pr-testing.yml"

This reverts commit d54eec419e2ddadb341804adde0e9b5792f02914.

Shriya-Chauhan committed:
feat: updated case studies page

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
fix: blog desktop design

Shriya-Chauhan committed:
fix: fixing mobile view

Shriya-Chauhan committed:
fix: changing heading sizes

Shriya-Chauhan committed:
fix: update ambassadors page

Shriya-Chauhan committed:
feat:adding graphic on ambassadors page

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
fix: tsc page draft

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
fix: roadmaps page redesign

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
fix: adding events and updates page

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
fix: lint errors

Shriya-Chauhan committed:
fix:add dark mode to individual blog post

Shriya-Chauhan committed:
fix: lint issues and warnings

Shriya-Chauhan committed:
fix:enhanced tools dashboard  page

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
fix:enhanced blog page card animation

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
fix: community page

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

ashmitjsg commented:
@Shriya-Chauhan Starting to review this, will complete the review by tomorrow.

Shriya-Chauhan committed:
fix:lint errors-s

ashmitjsg commented:
<img width="1919" height="1028" alt="image" src="https://github.com/user-attachments/assets/db41fe5c-3014-41b8-a3c7-95ba735d0610" />

The scrollbar should not appear at the bottom of the navmenu dropdown.

ashmitjsg commented:
On load, the dark mode should appear instantly instead of showing a transition from light mode:

https://github.com/user-attachments/assets/32be8528-a64b-4f43-808e-7919c6ba604f

ashmitjsg commented:
The font colour of the author's name in the  case study and the title of the TOC should be changed to something lighter in the dark mode.

Also, the colour of the left border of the TOC is very sharp as compared to background in the dark mode, please change it.

<img width="1919" height="534" alt="image" src="https://github.com/user-attachments/assets/fe5fd4af-1fc6-45ef-ab21-550218ac3653" />

ref: https://deploy-preview-4553--asyncapi-website.netlify.app/casestudies/adeogroup

ashmitjsg commented:
The go-to page dropdown on the case studies page is not working:

https://github.com/user-attachments/assets/4f031cf0-46d1-459e-b208-d5b7c77c59d8

ashmitjsg commented:
The pagination "go to page" dropdown should not go out of the screen, keep the height restricted to 3-4 options. Also, increase the width of the box and the dropdown to make the content fit correctly.

<img width="1919" height="1079" alt="image" src="https://github.com/user-attachments/assets/226456e2-73b8-46a4-bab7-1486101a5fa7" />

ref: https://deploy-preview-4553--asyncapi-website.netlify.app/blog

ashmitjsg commented:
The  hover background colour and the search icon color are too similar in dark mode, making it hard to recognise.

<img width="1919" height="200" alt="image" src="https://github.com/user-attachments/assets/f856856a-6ee3-43f9-9b03-681e8f3430bd" />

ashmitjsg commented:
The pagination number text and background colors are too similar in dark mode, making them hard to read.

Also, the "Go to page" dropdown box width is too small, the text and the dropdown icon are overlapping.

<img width="1885" height="508" alt="image" src="https://github.com/user-attachments/assets/40b79a2d-6ff1-4c37-8bae-05eb4afe9d1f" />

ashmitjsg commented:
On the roadmap page, the collapsible arrow and background colors are too similar in dark mode, making them hard to see.

<img width="458" height="263" alt="image" src="https://github.com/user-attachments/assets/5954aaf4-1c68-484b-9c2a-699dbaf87a44" />

ref: https://deploy-preview-4553--asyncapi-website.netlify.app/roadmap

ashmitjsg commented:
In the Events & Update Page, the event title text and background colors are too similar in dark mode, making them hard to read.

<img width="755" height="442" alt="image" src="https://github.com/user-attachments/assets/05016ebf-745b-4b1a-a07a-c85da7cc3e3f" />


ref: https://deploy-preview-4553--asyncapi-website.netlify.app/community/events-and-updates

ashmitjsg commented:
The Subscribe to Newsletter section is incomplete on the "Ambassadors" page. Ideally, you should use a single "Subscribe to Newsletter" component on all pages to avoid inconsistencies.

<img width="1906" height="547" alt="image" src="https://github.com/user-attachments/assets/80940fd7-eff7-4c43-9607-85e907bd3525" />

ref: https://deploy-preview-4553--asyncapi-website.netlify.app/community/ambassadors

ashmitjsg commented:
The community Dashboard page is incomplete for dark mode (headings, Hot Topics, Good First Issues) topics.

<img width="1919" height="1029" alt="image" src="https://github.com/user-attachments/assets/ce2e3390-54dc-4764-9928-18411467d3d7" />

ref: https://deploy-preview-4553--asyncapi-website.netlify.app/community/dashboard

ashmitjsg commented:
The community board committee page is incomplete.

<img width="1919" height="1034" alt="image" src="https://github.com/user-attachments/assets/1ec1271f-78e2-447b-b2eb-77f204b37879" />

ref: https://deploy-preview-4553--asyncapi-website.netlify.app/community/board

ashmitjsg commented:
In the parser tool page, the text color and background color are too similar in dark mode, making them hard to read.

<img width="1919" height="1028" alt="image" src="https://github.com/user-attachments/assets/65ae6178-309c-4e12-b399-dac906d9f10e" />

ref: https://deploy-preview-4553--asyncapi-website.netlify.app/tools/parsers


Same is the case with all tools pages:
- https://deploy-preview-4553--asyncapi-website.netlify.app/tools/generator
- https://deploy-preview-4553--asyncapi-website.netlify.app/en/tools/cli
- https://deploy-preview-4553--asyncapi-website.netlify.app/tools/modelina
- https://deploy-preview-4553--asyncapi-website.netlify.app/tools/github-actions

Shriya-Chauhan commented:
Hi @devilkiller-ag,

Since there were lots of code changes in this one. Can we start with one page review at a time whichever are mentioned in the description? 

Also, can we get the other pr merged first? Otherwise rebase would be difficult for me.


This one should be in draft for now. What do you think?

Shriya-Chauhan committed:
fix: added dark mode background colour to body

Shriya-Chauhan committed:
fix: dark mode smooth transition

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

ashmitjsg commented:
@coderabbit fullreview

ashmitjsg commented:
> Hi @devilkiller-ag,
> 
> Since there were lots of code changes in this one. Can we start with one page review at a time whichever are mentioned in the description?
 
That will take a lot of time. I have shared the per-page visual earlier. For the code part, I have triggered a full review by Coderabbit. In addition to that, I will go through all the files one by one.

> Also, can we get the other pr merged first? Otherwise rebase would be difficult for me.

The other PR was merged. Have you taken the latest pull from the base and master branches?

Shriya-Chauhan commented:
> > Hi @devilkiller-ag,
> > Since there were lots of code changes in this one. Can we start with one page review at a time whichever are mentioned in the description?
> 
> That will take a lot of time. I have shared the per-page visual earlier. For the code part, I have triggered a full review by Coderabbit. In addition to that, I will go through all the files one by one.
> 
> > Also, can we get the other pr merged first? Otherwise rebase would be difficult for me.
> 
> The other PR was merged. Have you taken the latest pull from the base and master branches?

yes

Shriya-Chauhan commented:
i have started making changes

> > > Hi @devilkiller-ag,
> > > Since there were lots of code changes in this one. Can we start with one page review at a time whichever are mentioned in the description?
> > 
> > 
> > That will take a lot of time. I have shared the per-page visual earlier. For the code part, I have triggered a full review by Coderabbit. In addition to that, I will go through all the files one by one.
> > > Also, can we get the other pr merged first? Otherwise rebase would be difficult for me.
> > 
> > 
> > The other PR was merged. Have you taken the latest pull from the base and master branches?
> 
> yes

I have started working on the changes.

Shriya-Chauhan committed:
fix: fixed dashboard

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan commented:
> On load, the dark mode should appear instantly instead of showing a transition from light mode:
> 
>  Screen.Recording.2025-11-13.224339.mp4

fixed

Shriya-Chauhan committed:
fix: lint errors

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan commented:
> The font colour of the author's name in the case study and the title of the TOC should be changed to something lighter in the dark mode.
> 
> Also, the colour of the left border of the TOC is very sharp as compared to background in the dark mode, please change it.
> 
> <img alt="image" width="1919" height="534" src="https://private-user-images.githubusercontent.com/43639341/514006330-fe5fd4af-1fc6-45ef-ab21-550218ac3653.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMzMDE2NjcsIm5iZiI6MTc2MzMwMTM2NywicGF0aCI6Ii80MzYzOTM0MS81MTQwMDYzMzAtZmU1ZmQ0YWYtMWZjNi00NWVmLWFiMjEtNTUwMjE4YWMzNjUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE2VDEzNTYwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5MDk0YmYzYWFlMmRmYTI5YzVhNjNjOGZiZGE2ZDY3Zjc3YzliNzc4OWY0MjljYWFkNzZjZGQxNzVmMmUzZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ionYB3yzYbGgitzqUzRvBv1iWQUtHRsjBXQ5O7NoWkc">
> ref: https://deploy-preview-4553--asyncapi-website.netlify.app/casestudies/adeogroup

fixed

Shriya-Chauhan committed:
fix: creating a single pagination component

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
fix: dark mode mode for board members page

Shriya-Chauhan committed:
fix: fixing community icons

Shriya-Chauhan committed:
fix: dark mode to tools pages

Shriya-Chauhan committed:
fix: dark mode to roadmaps arrows

Shriya-Chauhan committed:
fix: dark mode subscribe section

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
fix: added date data to blog cards

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

aeworxet commented:
Is there any news on this PR?
Keeping it unmerged is blocking @patilpratik1905's PR https://github.com/asyncapi/website/pull/4489.

princerajpoot20 commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,46 +1,95 @@
 
-[homepage]: http://contributor-covenant.org
-[version]: http://contributor-covenant.org/version/1/4/
+For answers to common questions about Contributor Covenant, see the FAQ at [https://www.contributor-covenant.org/faq](https://www.contributor-covenant.org/faq). Translations are provided at [https://www.contributor-covenant.org/translations](https://www.contributor-covenant.org/translations). Additional enforcement and community guideline resources can be found at [https://www.contributor-covenan ...
```
What is the reason behind making changes to the Code of Conduct?

Shriya-Chauhan replied:
not my changes, came from the master branch.

princerajpoot20 replied:
In that case, please update the target branch. 
`These changes cannot go ahead in this PR, as they are not part of your PR.`

Shriya-Chauhan replied:
i believe you have to sync it with master.

princerajpoot20 commented on components/CaseStudyCard.tsx:
```diff
@@ -1,16 +1,18 @@
 
 /**
- * @description A component that displays a list of case studies in a card format
+ * @description Enhanced case study card component with modern design
```
Please improve the description. ```It currently reads like AI-generated text```. 

```The description shouldn’t use terms like “Enhanced component” because this code will be viewed as it is, not compared with older versions — so words like “enhanced” become redundant.```


For documentation or comments in code, the description should explain something that is not already obvious from simply reading the code. Otherwise, keep it short and straightforward. You can refer to clean coding guidelines for better clarity on this.

Shriya-Chauhan replied:
got it

princerajpoot20 commented on components/CaseStudyCard.tsx:
```diff
@@ -19,21 +21,29 @@ export default function CaseStudyCard({ studies = [] }: ICaseStudyCardProps) {
               {study.company.description}
             </Paragraph>
+
+            {/* CTA Button */}
```
As I mentioned in the above review comment, avoid adding comments that state obvious things. Your code should be self-explanatory, which can be achieved by following clean coding practices.

princerajpoot20 commented on components/Pagination.tsx:
```diff
@@ -0,0 +1,271 @@
+        </button>
+      </div>
+
+      {/* Go to Page Dropdown */}
```
As I mentioned in the above review comment, avoid adding comments that state obvious things. Please refer to clean coding practices.

princerajpoot20 commented on components/community/TSCMemberCard.tsx:
```diff
@@ -0,0 +1,98 @@
+          >
+            <svg className='h-5 w-5' fill='currentColor' viewBox='0 0 20 20'>
+              <path d='M16.338 16.338H13.67V12.16c0-.995-.017-2.277-1.387-2.277-1.39 0-1.601 1.086-1.601 2.207v4.248H8.014v-8.59h2.559v1.174h.037c.356-.675 1.227-1.387 2.526-1.387 2.703 0 3.203 1.778 3.203 4.092v4.711zM5.005 6.575a1.548 1.548 0 11-.003-3.096 1.548 1.548 0 01.003 3.096zm-1.337 9.763H6.34v-8.59H3.667v8.59zM17.668 1H2.328C1.595 1 1 1.581 1 2.298v15.403C1 18.418 1.595 19 2.328 19h15. ...
+            </svg>
```
How about separating out these SVGs for better readability of this file?

Shriya-Chauhan replied:
works!

princerajpoot20 commented on components/dashboard/table/Table.tsx:
```diff
@@ -22,34 +22,68 @@ interface TableProps {
+                <li className='flex items-start gap-2'>
+                  <span className='text-primary-500 dark:text-primary-400 mt-1'>•</span>
+                  <span className='dark:text-gray-300'>
+                    Join our{' '}
+                    <a
+                      href='https://asyncapi.com/slack-invite'
+                      className='text-blue-500 dark:text-blue-400 underline font-medium'
+                    >
+                      Slack
+                    </a>{' '}
```
Do we really need this ```{' '}```? From what I recall, JSX will preserve the spaces around the <a> properly. If not, there must be a cleaner way to handle this.

Shriya-Chauhan replied:
Prettier requires {' '} on a new line.

princerajpoot20 commented on components/dashboard/table/Table.tsx:
```diff
@@ -22,34 +22,68 @@ interface TableProps {
+                  <span className='dark:text-gray-300'>
+                    In the <span className='font-semibold text-green-700 dark:text-green-400'>#11_contributing</span>{' '}
+                    channel, call out the maintainers that you want to work with. Ask them if there are any issues you
+                    could solve. You know who these people are from{' '}
```
Same here

princerajpoot20 commented on components/navigation/EventPostItem.tsx:
```diff
@@ -59,33 +45,46 @@ function EventPostItem({ post, className = '', id }: EventPostItemProps): React.
           </div>
-        </a>
+
+          {/* Button */}
```
As I mentioned earlier, please review all the comments and documentation changes you have made throughout the PR, and avoid adding comments that state obvious things. Please refer to clean coding practices.

princerajpoot20 commented on config/meetings.json:
```diff
@@ -1,17 +1,31 @@
     "title": "Marketing WG Meeting",
-    "calLink": "https://www.google.com/calendar/event?eid=Z2hrM2IzcTZma2FpbzlrNzJicDlxZjU1ZWMgY19xOXRzZWlnbG9tZHNqNm5qdWh2YnB0czExY0Bn",
-    "url": "https://github.com/asyncapi/community/issues/1968",
+    "calLink": "https://www.google.com/calendar/event?eid=MnNmMmtnbzI2MDdicXZmbWtsMnYyZWNrYTAgY19xOXRzZWlnbG9tZHNqNm5qdWh2YnB0czExY0Bn",
+    "url": "https://github.com/asyncapi/community/issues/1925",
     "banner": "",
-    "date": "2025-07-08T14:00:00.000Z"
+    "date": "2025-10-06T14:00:00.000Z"
   },
```
Just wondering about the reason behind this change. 

Also, is there any change in the Google Calendar event that I might be missing? I see a change in the event as well.

Shriya-Chauhan replied:
Can you pls sync this branch with master. this is not my change.

princerajpoot20 commented on config/meetings.json:
```diff
@@ -1,17 +1,31 @@
+  },
+  {
+    "title": "Governance Board Meeting",
+    "calLink": "https://www.google.com/calendar/event?eid=sample3",
```
It seems like a dummy URL that you’ve pasted here. Or am I missing something?

Shriya-Chauhan replied:
Dummy only.

Shriya-Chauhan replied:
removed it

princerajpoot20 commented on pages/blog/index.tsx:
```diff
@@ -61,71 +74,185 @@ export default function BlogIndexPage() {
+      if (window.innerWidth < 640) {
+        setPostsPerPage(5); // Mobile: 5 posts
+      } else {
+        setPostsPerPage(9); // Tablet/Desktop: 9 posts
```
Again, please follow clean coding practices.

princerajpoot20 commented on pages/blog/index.tsx:
```diff
@@ -61,71 +74,185 @@ export default function BlogIndexPage() {
+    window.addEventListener('resize', handleResize);
+
+    // Cleanup
+    return () => window.removeEventListener('resize', handleResize);
```
Again, please follow clean coding practices. Maybe we need to rename things for better clarity instead of adding comments to explain them.

princerajpoot20 commented on pages/casestudies/index.tsx:
```diff
@@ -23,128 +26,280 @@ interface Adopter {
+
 /**
- * @description Renders the Case Studies page.
+ * @description Renders the Case Studies page with modern design and horizontal card carousel.
```
Again, what does the term ```modern design``` convey? It’s just another BUZZ word.

princerajpoot20 commented on pages/community/ambassadors/index.tsx:
```diff
@@ -1,13 +1,26 @@
+import { HeadingLevel, HeadingTypeStyle } from '@/types/typography/Heading';
 
 import Button from '../../../components/buttons/Button';
+import IconBook from '../../../components/icons/Book';
+import IconClipboard from '../../../components/icons/Clipboard';
+import IconCode from '../../../components/icons/Code';
+import IconDocument from '../../../components/icons/Document';
+import IconLightbulb from '../../../components/icons/LightBulb';
+import IconMicrophone from '../../../components/icons/Microphone';
+import IconPlay from '../../../components/icons/Play';
+import IconStar from '../../../components/icons/Star';
+import IconTerminal from '../../../components/icons/Terminal';
+import IconUsersGroup from '../../../components/icons/UsersGroup';
+import IconVideo from '../../../components/icons/Video';
+import Container from '../../../components/layout/Container';
 import GenericLayout from '../../../components/layout/GenericLayout';
 import NewsletterSubscribe from '../../../components/NewsletterSubscribe';
+import PaginationComponent from '../../../components/Pagination';
```
These imports should be in ```alphabetical order for better readability.``` Please refer to clean coding practices.

princerajpoot20 commented on pages/community/ambassadors/index.tsx:
```diff
@@ -49,178 +81,376 @@ export default function Index() {
-              excitement within their communities to help others build better software.
+      </div>
+
+      {/* Contributions Section */}
```
Again, please follow clean coding practices everywhere.

princerajpoot20 commented on pages/community/ambassadors/index.tsx:
```diff
@@ -49,178 +81,376 @@ export default function Index() {
+              <Button
+                className='w-full sm:w-auto bg-white text-primary-600 hover:bg-gray-50 focus:outline-none font-semibold shadow-lg'
+                text='Become an Ambassador Now'
+                href='https://github.com/asyncapi/community/blob/master/AMBASSADOR_ORGANIZATION.md#are-you-interested-in-becoming-an-official-asyncapi-ambassador'
```
I would suggest avoiding all these ```hard-coded values``` directly in the code. These might change in the future, and as we know, the recent documentation changes in the community repo have already increased the maintenance workload tremendously. Hence, I would recommend creating an appropriate constants file and moving these values there.

Shriya-Chauhan replied:
not changing this for now.

princerajpoot20 commented on pages/community/events-and-updates.tsx:
```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+
+import Button from '@/components/buttons/Button';
+import GoogleCalendarButton from '@/components/buttons/GoogleCalendarButton';
+import ICSFileButton from '@/components/buttons/ICSFileButton';
+import SubscribeButton from '@/components/buttons/SubscribeButton';
+import Calendar from '@/components/Calendar';
+import FeatureCard from '@/components/community/FeatureCard';
+import IconCalendar from '@/components/icons/Calendar';
+import IconNewsroom from '@/components/icons/Newsroom';
+import IconUsersGroup from '@/components/icons/UsersGroup';
+import IconVideo from '@/components/icons/Video';
+import Container from '@/components/layout/Container';
+import GenericLayout from '@/components/layout/GenericLayout';
+import NewsletterSubscribe from '@/components/NewsletterSubscribe';
+import NewsroomBlogPosts from '@/components/newsroom/NewsroomBlogPosts';
+import NewsroomYoutube from '@/components/newsroom/NewsroomYoutube';
+import Heading from '@/components/typography/Heading';
+import Paragraph from '@/components/typography/Paragraph';
+import TextLink from '@/components/typography/TextLink';
+import { HeadingLevel, HeadingTypeStyle } from '@/types/typography/Heading';
+import { ParagraphTypeStyle } from '@/types/typography/Paragraph';
+import { makeStaticProps } from '@/utils/getStatic';
```
As mentioned earlier, these imports should be in alphabetical order for better readability. Please refer to clean coding practices.

princerajpoot20 commented on pages/community/index.tsx:
```diff
@@ -37,177 +29,371 @@ export default function CommunityIndexPage() {
+                    strokeWidth={2}
+                    d='M9 12l2 2 4-4M7.835 4.697a3.42 3.42 0 001.946-.806 3.42 3.42 0 014.438 0 3.42 3.42 0 001.946.806 3.42 3.42 0 013.138 3.138 3.42 3.42 0 00.806 1.946 3.42 3.42 0 010 4.438 3.42 3.42 0 00-.806 1.946 3.42 3.42 0 01-3.138 3.138 3.42 3.42 0 00-1.946.806 3.42 3.42 0 01-4.438 0 3.42 3.42 0 00-1.946-.806 3.42 3.42 0 01-3.138-3.138 3.42 3.42 0 00-.806-1.946 3.42 3.42 0 010-4.438 3.42 ...
+                  />
+                </svg>
```
Avoid adding the SVG directly here. Extract it to an appropriate location. This is making the file bulky and heavily impacting readability.

Also, there should not be any SVGs directly in the code file. ```This heavily impacts the readability of the code, and they have nothing to do with the actual logic. These are just static values that no one cares about when going through the code.``` Hence, please extract all the SVGs to an appropriate location and import them wherever needed. This also improves reusability and, of course, readability as mentioned.

princerajpoot20 commented on pages/community/index.tsx:
```diff
@@ -37,177 +29,371 @@ export default function CommunityIndexPage() {
+                    strokeWidth={2}
+                    d='M9 12l2 2 4-4M7.835 4.697a3.42 3.42 0 001.946-.806 3.42 3.42 0 014.438 0 3.42 3.42 0 001.946.806 3.42 3.42 0 013.138 3.138 3.42 3.42 0 00.806 1.946 3.42 3.42 0 010 4.438 3.42 3.42 0 00-.806 1.946 3.42 3.42 0 01-3.138 3.138 3.42 3.42 0 00-1.946.806 3.42 3.42 0 01-4.438 0 3.42 3.42 0 00-1.946-.806 3.42 3.42 0 01-3.138-3.138 3.42 3.42 0 00-.806-1.946 3.42 3.42 0 010-4.438 3.42 ...
+                  />
+                </svg>
```
As mentioned above, the only reason this file is bulky and unreadable is because of these SVGs.

princerajpoot20 commented on pages/community/tsc.tsx:
```diff
@@ -1,65 +1,319 @@
+import { useState } from 'react';
+
+import type { Tsc } from '@/types/pages/community/Community';
+
+import Button from '../../components/buttons/Button';
+import TSCMemberCard from '../../components/community/TSCMemberCard';
+import IconArrowRight from '../../components/icons/ArrowRight';
+import IconDocument from '../../components/icons/Document';
+import IconUsersGroup from '../../components/icons/UsersGroup';
+import GenericLayout from '../../components/layout/GenericLayout';
+import NewsletterSubscribe from '../../components/NewsletterSubscribe';
+import PaginationComponent from '../../components/Pagination';
+import tscBoardList from '../../config/TSC_BOARD_MEMBERS.json';
```
As mentioned earlier, these imports should be in alphabetical order for better readability. Please refer to clean coding practices.

princerajpoot20 commented on pages/community/tsc.tsx:
```diff
@@ -1,65 +1,319 @@
+  // add avatar url
+  userData.avatarUrl = `${userData.githubUrl}.png`;
+
+  // make repo links
```
Again, remove self-explanatory comments and unnecessary spacing.

princerajpoot20 commented on pages/tools/index.tsx:
```diff
@@ -17,26 +17,98 @@ export default function ToolsIndex() {
+                    <path
+                      fillRule='evenodd'
+                      d='M11.3 1.046A1 1 0 0112 2v5h4a1 1 0 01.82 1.573l-7 10A1 1 0 018 18v-5H4a1 1 0 01-.82-1.573l7-10a1 1 0 011.12-.38z'
+                      clipRule='evenodd'
```
As mentioned above, extract these out from all places.

princerajpoot20 commented on tsconfig.tsbuildinfo:
```diff
@@ -0,0 +1 @@
+{"fileNames":["./node_modules/typescript/lib/lib.es5.d.ts","./node_modules/typescript/lib/lib.es2015.d.ts","./node_modules/typescript/lib/lib.es2016.d.ts","./node_modules/typescript/lib/lib.es2017.d.ts","./node_modules/typescript/lib/lib.es2018.d.ts","./node_modules/typescript/lib/lib.es2019.d.ts","./node_modules/typescript/lib/lib.es2020.d.ts","./node_modules/typescript/lib/lib.es2021.d.ts","./n ...
```
Can you please explain the relevance of this file? What purpose does it serve? Do we really need this in our repo, or can it be added to .gitignore?

Shriya-Chauhan replied:
we should  its a build artifact not source code. I believe its coming from 
<img width="280" height="98" alt="image" src="https://github.com/user-attachments/assets/29c8d9ee-5e97-4b1b-ba41-b3efe00cc221" />
 in tscnfig.json. What do you suggest? @princerajpoot20

princerajpoot20 commented on types/components/navigation/BlogPostType.ts:
```diff
@@ -2,5 +2,8 @@ export enum BlogPostType {
   Marketing = 'marketing',
   Strategy = 'strategy',
-  Communication = 'communication'
+  Communication = 'communication',
+  Conference = 'conference',
+  Engineering = 'engineering',
+  Community = 'community'
```
Again, as mentioned earlier, these should be in alphabetical order for better readability. Please refer to clean coding practices.

princerajpoot20 commented on types/pages/community/Community.ts:
```diff
@@ -50,4 +50,8 @@ export interface Tsc {
   githubID: number
   isBoardMember?: boolean;
   isBoardChair?: boolean;
+  githubUrl?: string;
```
Do we really need this githubUrl? I can see there is already a GitHub ID. What is the purpose of this extra, redundant data?

princerajpoot20 commented on types/pages/community/Community.ts:
```diff
@@ -50,4 +50,8 @@ export interface Tsc {
+  githubUrl?: string;
+  twitterUrl?: string;
+  linkedinUrl?: string;
+  reposList?: { name: string; url: string }[];
```
Umm… I see these attributes are also already present above. Can you explain the reason for having them again?

Shriya-Chauhan replied:
was testing.

princerajpoot20 reviewed (changes requested):
Hi @Shriya-Chauhan, please look into the comments. 

```The review is not completed yet,``` as it is such a big PR to go through, but you can start working on the current comments. Also, please go through the CodeRabbit suggestions. 

I’ll continue the review over the coming weekend. Thanks.

princerajpoot20 commented:
@aeworxet This PR will take time. 

It’s quite a huge PR, and ```even reviewing a small percentage of it is taking hours.``` From my point of view, we should not rush and compromise on quality, especially when there is no reason to hurry. So please act accordingly.

princerajpoot20 commented on components/community/ToolingsShowcase.tsx:
```diff
@@ -0,0 +1,186 @@
+/**
+ * @description ToolingsShowcase component for displaying AsyncAPI tools collection
+ */
+export default function ToolingsShowcase() {
```
There is no such word as `Toolings`, as it’s an uncountable noun. Please have a look at this.

princerajpoot20 commented on components/community/ToolingsShowcase.tsx:
```diff
@@ -0,0 +1,186 @@
+    badge: 'Open Source'
+  }
+};
+
```
Avoid keeping this `hardcoded data` inside the component itself. 

It makes the component unnecessarily bulky, the data becomes non-reusable, ```and most importantly, it becomes very difficult to maintain.```

We usually keep this kind of data in a JSON file. Please extract it out.

Also, can you please confirm where you got this data from? I mean, is this data already shown on our current website?
If yes, can you please share the reference?
If no, then we need to review this data to decide which tools we want to display on the website.

princerajpoot20 commented on components/community/ToolingsShowcase.tsx:
```diff
@@ -0,0 +1,186 @@
+'use client';
```
Umm… what is the purpose or reason behind this line?

Shriya-Chauhan replied:
component is u=interactive, so using hooks and event handlers

Shriya-Chauhan commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,46 +1,95 @@
+
+- Join our [AsyncAPI Slack](https://asyncapi.com/slack-invite) and share your report in the `#coc` channel.  
+- Reach out directly to any member of the [Code of Conduct Committee](https://github.com/orgs/asyncapi/teams/code_of_conduct).
+- Or, if you’d prefer, just send us an email at **conduct@asyncapi.com**.
```
this hasnt been changed by me and comes from the master branch

Shriya-Chauhan commented on components/community/ToolingCard.tsx:
```diff
@@ -0,0 +1,117 @@
+          <div className='absolute -top-16 -right-24 w-[500px] h-[280px] bg-gradient-to-br from-blue-100 to-blue-200 dark:from-blue-900/30 dark:to-blue-800/30 opacity-40 transform rotate-[25deg] rounded-[70px]' />
+          <div className='absolute -top-12 -right-16 w-[420px] h-[260px] bg-gradient-to-br from-purple-100 to-purple-200 dark:from-purple-900/30 dark:to-purple-800/30 opacity-50 transform rotate-[25deg] rounded-[60px]' />
+
+          {/* Bottom Large Fan/Petal Shape - Blue gradient */}
+          <div className='absolute -bottom-40 -left-32 w-[700px] h-[500px]'>
+            <svg viewBox='0 0 500 500' xmlns='http://www.w3.org/2000/svg' className='w-full h-full'>
+              <defs>
+                <linearGradient id='blueGradient' x1='20%' y1='80%' x2='80%' y2='20%'>
+                  <stop
+                    offset='0%'
+                    className='text-primary-500 dark:text-primary-600'
+                    stopColor='currentColor'
+                    stopOpacity='0.9'
+                  />
+                  <stop
+                    offset='40%'
+                    className='text-blue-400 dark:text-blue-500'
+                    stopColor='currentColor'
+                    stopOpacity='0.85'
+                  />
+                  <stop
+                    offset='100%'
+                    className='text-cyan-400 dark:text-cyan-600'
+                    stopColor='currentColor'
+                    stopOpacity='0.8'
+                  />
+                </linearGradient>
+              </defs>
+              {/* Fan/Petal shape spreading from bottom left */}
+              <path
+                fill='url(#blueGradient)'
+                d='M 50 450 Q 150 350 280 320 Q 400 300 480 340 Q 500 380 470 420
+                   Q 440 460 380 480 Q 280 500 180 490 Q 100 480 50 450 Z
+                   M 60 440 Q 140 380 240 360 Q 320 350 380 380
+                   L 320 420 Q 240 400 160 430 Z
+                   M 80 430 Q 150 390 220 380 Q 280 375 320 395
+                   L 280 420 Q 220 410 160 425 Z'
+              />
+            </svg>
+          </div>
```
done

Shriya-Chauhan committed:
fix: fixing code rabbit suggestions

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
refactor: address code review comments

- Improve JSDoc descriptions and remove obvious comments
- Extract SVG icons in TSCMemberCard into separate components
- Reduce unnecessary JSX spacing expressions
- Improve code clarity per clean coding practices

Signed-off-by: Shriya-Chauhan <auroralflower@gmail.com>

Shriya-Chauhan committed:
refactor: remove dymmy data

Shriya-Chauhan committed:
test

Shriya-Chauhan committed:
test

Shriya-Chauhan committed:
fixing svgs

princerajpoot20 commented:
@Shriya-Chauhan I have merged the `master` branch into `website-ui`. You can now resolve the merge conflict. 
Let me know when it’s ready for review.

princerajpoot20 commented:
@Shriya-Chauhan Ping pong. any update on this one?

Shriya-Chauhan commented:
> @Shriya-Chauhan Ping pong. any update on this one?

Hello, will make changes about the last few reviews

Shriya-Chauhan committed:
fix: adding svgs as seperate files

Shriya-Chauhan committed:
Remove tsconfig.tsbuildinfo from version control

Shriya-Chauhan committed:
refactor(community): remove redundant computed fields from types

Remove computed URL fields (githubUrl, twitterUrl, linkedinUrl, avatarUrl, reposList)
from Tsc and Ambassador interfaces. Compute URLs inline in components instead of
storing them in type definitions. This improves type accuracy and reduces redundancy.

Shriya-Chauhan committed:
refactor: seprated data rom components

Shriya-Chauhan committed:
fix: add missing isBoardChair to Ambassador interface and fix pagination import

Shriya-Chauhan committed:
fix: lint errors - import sorting and prettier formatting

princerajpoot20 commented on components/icons/LightBulb.tsx:
```diff
@@ -1,20 +1,10 @@
+ * @description Icon component for Light Bulb
  */
-export default function IconLightBulb({ className = '' }) {
+export default function IconLightbulb({ className = '' }) {
```
Nit: Unnecessary change. Also, it does not follow the casing convention

Shriya-Chauhan replied:
Fixed the dimension for the icons, thats why.

princerajpoot20 commented on components/icons/LightBulb.tsx:
```diff
@@ -1,20 +1,10 @@
-    >
-      <path d='M13 24h-2c-.288 0-.563-.125-.753-.341l-.576-.659h4.658l-.576.659c-.19.216-.465.341-.753.341zm1.867-3c.287 0 .52.224.52.5s-.233.5-.52.5h-5.734c-.287 0-.52-.224-.52-.5s.233-.5.52-.5h5.734zm-2.871-17c2.983 0 6.004 1.97 6.004 5.734 0 1.937-.97 3.622-1.907 5.252-.907 1.574-1.843 3.201-1.844 5.014h1.001c0-3.286 3.75-6.103 3.75-10.266 0-4.34-3.502-6.734-7.004-6.734-3.498 0-6.996 2.391-6.9 ...
+    <svg className={className} fill='currentColor' viewBox='0 0 20 20' xmlns='http://www.w3.org/2000/svg'>
+      <path d='M11 3a1 1 0 10-2 0v1a1 1 0 102 0V3zM15.657 5.757a1 1 0 00-1.414-1.414l-.707.707a1 1 0 001.414 1.414l.707-.707zM18 10a1 1 0 01-1 1h-1a1 1 0 110-2h1a1 1 0 011 1zM5.05 6.464A1 1 0 106.464 5.05l-.707-.707a1 1 0 00-1.414 1.414l.707.707zM5 10a1 1 0 01-1 1H3a1 1 0 110-2h1a1 1 0 011 1zM8 16v-1h4v1a2 2 0 11-4 0zM12 14c.015-.34.208-.646.477-.859a4 4 0 10-4.954 0c.27.213.462.519.476.859h4.002 ...
```
Hmm, I’m not sure about this one. Do we really need this change? It seems to modify the existing light theme.

Shriya-Chauhan replied:
explained this above

princerajpoot20 commented on components/icons/CommunityDriven.tsx:
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+/* eslint-disable max-len */
```
Do we really need this? If yes, I think we should reduce the length. 
Also, have a look at other similar comments as well.

Shriya-Chauhan replied:
Yes

princerajpoot20 commented:
FYI: Had a discussion with @Shriya-Chauhan. For now, we are merging this PR. We will do the remaining sanity checks and refactoring after this, and she will raise a follow-up PR in the `website-ui` branch.

Will create a follow-up issue to keep track of the remaining items and also start planning the rollout to the master branch. But for now, we can merge this.

Good work @Shriya-Chauhan!!
